### PR TITLE
[processor/attributes] Add support to filter on log body

### PR DESCRIPTION
**Pull request by atoulme**
**Description:** 
Add support to filter logs based on their body.

**Link to tracking Issue:**
None

**Testing:** 
Unit tests

**Documentation:**
Added to README

### CODE REVIEW

**Review by pmm-sumo — COMMENTED** [review]
I think example in README.md how to use the `log_bodies` would be good to have. Also one in [testdata](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor/testdata)

I believe we should a add a positive test case, perhaps [basing on TestLogRecord_Matching_True](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/coreinternal/processor/filterlog/filterlog_test.go#L131)

This change should be mentioned in changelog

@open-telemetry/log-collection-approvers could you take a look as well?

**Comment by sumo-drosiek** [comment]
Could it support regexes?

**Comment by pmm-sumo** [comment]
> Could it support regexes?

I believe so, since it's using [filterset](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/coreinternal/processor/filterset/config.go#L41-L44) underneath, which can be also configured with regular expression to match

**Comment by sumo-drosiek** [comment]
also, kind of related PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5680

**Comment by atoulme** [comment]
@sumo-drosiek your PR looks much better than mine, why didn't it make it? My PR is just about one thing, getting log bodies to be used for the processor. Right now, the processor is useless for logs since it only allows to filter on the name attribute (I actually think looking at the logs the name filters is not working) and that attribute is deprecated.

So if that small scope of change is ok and gets us there, please don't mind me and I'll keep plugging away.

**Comment by pmm-sumo** [comment]
> @sumo-drosiek your PR looks much better than mine, why didn't it make it? My PR is just about one thing, getting log bodies to be used for the processor. Right now, the processor is useless for logs since it only allows to filter on the name attribute (I actually think looking at the logs the name filters is not working) and that attribute is deprecated.
> 
> So if that small scope of change is ok and gets us there, please don't mind me and I'll keep plugging away.

I believe there was some research on using common expression language for OpenTelemetry Collector components in progress back then, which I believe [ended up with conclusion](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5887#issuecomment-1025946362) that we want to follow with what is now [transformprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor). 

The latter is still a work in progress, so I think we should be OK with making some small changes/additions to filter/attributes/resource processors as needed (like adding another fields which can be filtered), though I believe that adding another expression language here is too much and we should rather wait until `transformprocessor` is ready and supports more signals. Please correct me if I'm wrong here @bogdandrutu @anuraaga :)

**Comment by anuraaga** [comment]
That sounds about right. This is SlowTel so I'd recommend not expecting quick progress regardless of the component ;)

**Comment by atoulme** [comment]
@pmm-sumo feedback addressed, but no example in README, sorry. I didn't see a good spot for it.

File: processor/attributesprocessor/README.md [hunk 1]
```diff
@@ -218,6 +218,10 @@ attributes:
       # This is an optional field.
       log_names: [<item1>, ..., <itemN>]
 
+      # The log body must match at least one of the items.
+      # This is an optional field.
+      log_bodies: [<item1>, ..., <itemN>]
```
Comment (pmm-sumo) [inline processor/attributesprocessor/README.md:223]: I think it's worth mentioning that only String bodies are matched against here

**Review by pmm-sumo — APPROVED** [review]

File: processor/attributesprocessor/README.md [hunk 2]
```diff
@@ -218,6 +218,10 @@ attributes:
       # This is an optional field.
       log_names: [<item1>, ..., <itemN>]
 
+      # The log body must match at least one of the items.
```
Comment (pmm-sumo) [inline processor/attributesprocessor/README.md:221]:
```suggestion
      # The log body must match at least one of the items.
      # Currently only string body types are supported.
```

**Review by dmitryax — APPROVED** [review]

**Comment by sumo-drosiek** [comment]
@atoulme I just linked it because it is related, but as @pmm-sumo the conclusion was to skip adding expr language :)

**Review by djaglowski — APPROVED** [review]

**Pull request merged**